### ISSUE TITLE
When you click on the menu, it will automatically click on the menu item

### ISSUE THREAD

**Issue by o-3-o** [ocornut/imgui#8233]
### Version/Branch of Dear ImGui:

Version 1.91.6 master

### Back-ends:

imgui_impl_dx11.cpp + imgui_impl_win32.cpp

### Compiler, OS:

win11+vs2022

### Full config/build information:

```
Dear ImGui 1.91.6 (19160)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1942
define: _MSVC_LANG=201402
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx11
io.ConfigFlags: 0x00000003
 NavEnableKeyboard
 NavEnableGamepad
io.ConfigNavCaptureKeyboard
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000000E
 HasMouseCursors
 HasSetMousePos
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,128
io.DisplaySize: 1264.00,761.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

### Details:

If the mouse is over a sub-item when the menu pops up, it will automatically click

### Screenshots/Video:

![动画](https://github.com/user-attachments/assets/bdd66de2-74ad-4cb0-9fac-b878d6893400)

### Minimal, Complete and Verifiable Example code:

```cpp
ImGui::ShowDemoWindow();
```

**Comment by ocornut** [comment]
Thanks for your report.
This is tricky. `MenuItem()` intentionally uses `ImGuiSelectableFlags_NoSetKeyOwner`:

```cpp
// We use ImGuiSelectableFlags_NoSetKeyOwner to allow down on one menu item, move, up on another.
const ImGuiSelectableFlags selectable_flags = ImGuiSelectableFlags_SelectOnRelease | ImGuiSelectableFlags_NoSetKeyOwner | ImGuiSelectableFlags_SetNavIdOnHover;
```

The reason is that we explicitly want a single mouse down to be able to browse child menus (some people use menus this way).
But that logic breaks when the next appearing child menu is overlapping the current mouse position.
I don't yet know what the better workaround would be.

Are you directly impacted by this?

**Comment by o-3-o** [comment]
> Thanks for your report. This is tricky. `MenuItem()` intentionally uses `ImGuiSelectableFlags_NoSetKeyOwner`:
> 
> ```c++
> // We use ImGuiSelectableFlags_NoSetKeyOwner to allow down on one menu item, move, up on another.
> const ImGuiSelectableFlags selectable_flags = ImGuiSelectableFlags_SelectOnRelease | ImGuiSelectableFlags_NoSetKeyOwner | ImGuiSelectableFlags_SetNavIdOnHover;
> ```
> 
> The reason is that we explicitly want a single mouse down to be able to browse child menus (some people use menus this way). But that logic breaks when the next appearing child menu is overlapping the current mouse position. I don't yet know what the better workaround would be.
> 
> Are you directly impacted by this?

Thank you for your answer. Yes, I did solve this problem after commenting out `if (flags & ImGuiSelectableFlags_NoSetKeyOwner) { button_flags |= ImGuiButtonFlags_NoSetKeyOwner; }` in the `Selectable` API. However, the `BeginMenu` API does not set an option for this flag, which will result in the inability to click on the expected menu item when the window size is small.

**Comment by o-3-o** [comment]
Hello @ocornut  Excuse me, is there a way to prevent the menu from using the ImGuiSelectableFlags_NoSetKeyOwner flag? Without polluting the Dear ImGui file.

**Comment by ocornut** [comment]
Not yet but I will investigate the issue.

Out of clarity, can you clarify in which way it is impacting you in real conditions? As i imagine your os window is not as small in reality.

**Comment by o-3-o** [comment]
Yes, I want to use imgui to make a desktop application UI framework, but it is still in the testing stage of the framework. This desktop application fills the main window with full screen. Under normal circumstances, this problem is not easy to trigger. In order to deal with all emergencies, it does not rule out that other users will set the window to this size. It would be perfect if there is an option to control it.
![Uploading 动画.gif…]()

**Comment by ocornut** [comment]
Pushed a fix bca5a69 for this issue now!
